NetCasbin/Enforcer.cs

```diff
@@ -79,177 +87,248 @@ public List<string> GetUsersForRoles(string[] names)
+            var rules = role.Select(r => new[] {user, r}.Concat(domain).ToList());
+            return AddGroupingPolicies(rules);
+        }
+
+        /// <summary>
+        /// AddRolesForUser adds roles for a user.
+        /// </summary>
+        /// <param name="user"></param>
+        /// <param name="role"></param>
+        /// <param name="domain"></param>
         /// <returns>Returns false if the user already has the role (aka not affected).</returns>
-        public Task<bool> AddRoleForUserAsync(string user, string role)
+        public Task<bool> AddRolesForUserAsync(string user, IEnumerable<string> role, params string[] domain)
         {
-            return AddGroupingPolicyAsync(user, role);
+            var rules = role.Select(r => new[] {user, r}.Concat(domain).ToList());
+            return AddGroupingPoliciesAsync(rules);
         }
 
+        #endregion
+
+        #region Delete roles or users
+
         /// <summary>
         /// Deletes a role for a user.
         /// </summary>
         /// <param name="user"></param>
         /// <param name="role"></param>
+        /// <param name="domain"></param>
         /// <returns>Returns false if the user does not have the role (aka not affected).</returns>
-        public bool DeleteRoleForUser(string user, string role)
+        public bool DeleteRoleForUser(string user, string role, params string[] domain)
         {
-            return RemoveGroupingPolicy(user, role);
+            var parameters = new[] { user, role }.Concat(domain);
+            return RemoveGroupingPolicy(parameters.ToList());
         }
 
         /// <summary>
         /// Deletes a role for a user.
         /// </summary>
         /// <param name="user"></param>
         /// <param name="role"></param>
+        /// <param name="domain"></param>
         /// <returns>Returns false if the user does not have the role (aka not affected).</returns>
-        public Task<bool> DeleteRoleForUserAsync(string user, string role)
+        public Task<bool> DeleteRoleForUserAsync(string user, string role, params string[] domain)
         {
-            return RemoveGroupingPolicyAsync(user, role);
+            var parameters = new[] { user, role }.Concat(domain);
+            return RemoveGroupingPolicyAsync(parameters.ToList());
         }
 
         /// <summary>
         /// Deletes all roles for a user.
         /// </summary>
         /// <param name="user"></param>
+        /// <param name="domain"></param>
         /// <returns>Returns false if the user does not have any roles (aka not affected).</returns>
-        public bool DeleteRolesForUser(string user)
+        public bool DeleteRolesForUser(string user, params string[] domain)
         {
-            return RemoveFilteredGroupingPolicy(0, user);
+            string[] fieldValues = domain.Length switch
+            {
+                0 => new[] {user},
+                > 1 => throw new InvalidOperationException("Domain should be 1 parameter."),
```

Comment:
if domain length can't > 1, why use a string[] domain, how about using a optional parameter

Reply:
@Sagilio Go Casbin uses []string domain because it's the only way to represent optional parameter in Golang. I think it's unnecessary in C#. Just use a single optional string arg.

Reply:
Ok